tests/kokkos/kokkos_utils_tests.cpp

```diff
@@ -194,4 +194,61 @@ TEST_CASE("team_utils_large_ni", "[kokkos_utils]")
   test_utils_large_ni(.5);
 }
 
+template<typename Scalar, int length, bool Serialize>
+void test_parallel_reduce()
+{
+  using Device = ekat::DefaultDevice;
+  using MemberType = typename ekat::KokkosTypes<Device>::MemberType;
+  using ExeSpace = typename ekat::KokkosTypes<Device>::ExeSpace;
+
+#ifdef KOKKOS_ENABLE_OPENMP
+  const int n = omp_get_max_threads();
+  // test will not work with more than 16 threads
+  if (n > 16) {
+    WARN("Skipped because this test doesn't support more than 16 threads");
+    return;
+  }
+#endif
+
+  Kokkos::View<Scalar*, ExeSpace> data("data", length);
+  const auto data_h = Kokkos::create_mirror_view(data);
+  auto raw = data_h.data();
+  for (int i = 0; i < length; ++i)
+    raw[i] = 1.0/(i+1);
+  Kokkos::deep_copy(data, data_h);
+
+  Kokkos::View<Scalar*> results ("results", 1);
+  const auto results_h = Kokkos::create_mirror_view(results);
+
+  Scalar serial_result = Scalar();
+  for (int i = 0; i < length; ++i)
+    serial_result += 1.0/(i+1);
+
+  const auto policy =
+    ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, length);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+    Scalar team_result = Scalar();
+
+    const int begin = 0;
+    const int end = length;
+    ekat::ExeSpaceUtils<ExeSpace>::parallel_reduce<Serialize>(team, begin, end,
+        [&] (const int k, Scalar& reduction_value) {
+              reduction_value += data[k];
+        }, team_result);
+
+      results(0) = team_result;
+    });
+
+  Kokkos::deep_copy(results_h, results);
+  if (Serialize) {
+    REQUIRE(results_h(0) == serial_result);
+  }
```

Comment:
Would be good, if not Serialize, to have a tol-based requirement, something like `<= 10*std::numeric_limits<Scalar>::epsilon()`.

Reply:
What about using a sum that can be done exactly in FP arithmetic? Like `1+0.5+0.25+...+0.5^N`?

Reply:
Or just do it over int instead of FP.

Reply:
I like this tolerance approach, as it exercises the behavior of a reduction under "normal" non-BFB circumstances.

Reply:
The reason I went with a sum that is not bfb-exact was to just verify that the serial algorithm did the correct thing. Then using tolerance for parallel is just more convenient so that  I don't need to implement 2 version of the test.

Reply:
I guess in this case (with fixed inputs) a tol approach is just fine, since you can find a tol that is guaranteed to work. With random inputs, we got bitten in the past for running sometimes into the unlucky case where 1ulp propagates up the mantissa (in case of triggering several carries). So I'm fine with this.